go/erasablekv/erasable_kv_store.go

```diff
@@ -15,10 +17,23 @@ import (
 	"golang.org/x/net/context"
 )
 
+type UnboxError struct {
+	msg string
+}
+
+func NewUnboxError(msg string) *UnboxError {
+	return &UnboxError{msg: msg}
+}
+
+func (e UnboxError) Error() string {
+	return fmt.Sprintf("Unable to unbox item: %v", e.msg)
+}
+
 type boxedData struct {
 	V int
 	N [libkb.NaclDHNonceSize]byte
 	E []byte
+	H []byte // sha256 has of noise used for encryption
```

Comment:
This only gets used when there is an error right? So there is no downside for it being blank on current installs?

Reply:
yeah it's only for the error message and if someone is unable to unbox in the wild we can't add it in since we can't decrypt. I don't think it's worth trying to add to current keys since these code changes will allow them to participate in ek gen and after a week new clients will report if this is the issue.